### PR TITLE
JAMES-2628 Migrate CassandraTableManager to Reactor

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java [hunk 1]
```diff
@@ -60,27 +59,26 @@ public CassandraTableManager ensureAllTables() {
...
         CassandraAsyncExecutor executor = new CassandraAsyncExecutor(session);
-        FluentFutureStream.of(
-            module.moduleTables()
-                .stream()
+        Flux.fromIterable(module.moduleTables())
                 .map(CassandraTable::getName)
-                .map(name -> truncate(executor, name)))
-            .join();
+                .flatMap(name -> truncate(executor, name))
+                .publishOn(Schedulers.elastic())
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: you probably have to declare the `publishOn` before the flatmap.
Did you check with log ?

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java [hunk 2]
```diff
@@ -60,27 +59,26 @@ public CassandraTableManager ensureAllTables() {
...
 
-    private CompletableFuture<?> truncate(CassandraAsyncExecutor executor, String name) {
-        return executor.execute(
-            QueryBuilder.select()
-                .from(name)
-                .limit(1)
-                .setFetchSize(1))
-            .thenCompose(resultSet -> truncateIfNeeded(executor, name, resultSet));
+    private Mono<?> truncate(CassandraAsyncExecutor executor, String name) {
+        return Mono.fromFuture(executor.execute(
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: what about:
```java
Mono.fromFuture(...)
    .filter(ResultSet::isExhausted)
    .map(ignored -> Mono.fromFuture(executor.execute(QueryBuilder.truncate(name))))
```

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java (hunk 1 above)
Comment (blackheaven) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: Nothing in logs.

**Comment by trantienduchn** [comment]
```
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [INFO] Results:
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [INFO] 
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR] Failures: 
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.attachmentsShouldBeRetrievedWhenChainingSetMessagesAndGetMessagesTextAttachment Multiple Failures (2 failures)
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] 	Failed to parse the JSON document
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] 	com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /192.168.0.1:46981 (com.datastax.driver.core.exceptions.OperationTimedOutException: [/192.168.0.1:46981] Timed out waiting for server response))
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR] Errors: 
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   ForwardIntegrationTest.setUp:77 » Runtime javax.management.InstanceAlreadyExis...
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   ProvisioningTest.setup:63 » Runtime javax.management.InstanceAlreadyExistsExce...
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.setup:163 » Provision Unable to provision, see the follo...
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.setMessagesShouldMoveMessageInSentWhenMessageIsSent:2383 » ConditionTimeout
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.setMessagesShouldTriggerMaxQuotaReachedWhenTryingToSendMessageAndQuotaReached » Completion
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.setup:163 » Runtime javax.management.InstanceAlreadyExis...
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR]   SetMessagesMethodTest.setMessagesShouldUpdateKeywordsWhenKeywordsArePassed:449 » ConditionTimeout
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [INFO] 
[8aa1f545d5749912bf1d2733ea689b6eda8c1df3] [ERROR] Tests run: 18, Failures: 1, Errors: 7, Skipped: 0
```

**Comment by trantienduchn** [comment]
test this please

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java [hunk 3]
```diff
@@ -60,27 +58,21 @@ public CassandraTableManager ensureAllTables() {
...
-        FluentFutureStream.of(
-            module.moduleTables()
-                .stream()
+        Flux.fromIterable(module.moduleTables())
+                .publishOn(Schedulers.elastic())
                 .map(CassandraTable::getName)
-                .map(name -> truncate(executor, name)))
-            .join();
+                .flatMap(name -> truncate(executor, name))
+                .blockLast();
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: we don't rely want the last value, we want the Flux to complete.
You can use `then().block()` instead, it closer to you intention

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java (hunk 1 above)
Comment (mbaechler) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: what does it mean ? the logs should tell you where things are run

Comment (blackheaven) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: Either the logs may not have been activated, or I do not know what to look for.

Comment (mbaechler) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java]: I explained that yesterday:
* put `.log` at the right locations
* configure logback to output the logger with name="reactor" in DEBUG

**Pull request merged**